### PR TITLE
Optimize some command.

### PR DESCRIPTION
**Pull request by QiE2035**
1. Maybe a better TeleportCommand.
2. Add a TranslateCommand possibly useful.
3. Update README.md and README_zh-CN.md.
4. Maybe need to regulate.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -132,8 +132,9 @@ There is a dummy user named "Server" in every player's friends list that you can
...
 | talent         | talent <talentID> <value>                         | player.settalent          | Client only  | Set talent level for your current active character           |                                                 |
-| teleport       | teleport <x> <y> <z>                              | player.teleport           | Client only  | Change the player's position.                                | tp                                              |
+| teleport       | teleport [UID] <x\|~rx> <y\|~ry> <z\|~rz>         | player.teleport           | Both side    | Teleport a player to the position. Use ~ to use relative position.| tp                                         |
 | weather        | weather <weatherID> <climateID>                   | player.weather            | Client only  | Changes the weather                                          | w                                               |
+| translate      | translate [UID] <distance>                        | player.translate          | Both side    | Translate a player forward to the distance.                  | tr                                              |
```
Comment (alphagocc) [inline README.md:137]: Why `translate` means move forward？

File: src/main/java/emu/grasscutter/command/commands/TranslateCommand.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import emu.grasscutter.Grasscutter;
+import emu.grasscutter.command.Command;
+import emu.grasscutter.command.CommandHandler;
+import emu.grasscutter.game.player.Player;
+import emu.grasscutter.server.packet.send.PacketSceneEntityAppearNotify;
+import emu.grasscutter.utils.Position;
+
+import java.util.List;
+
+@Command(label = "trnslate", usage = "translate [UID] <distance>",
```
Comment (alphagocc) [inline src/main/java/emu/grasscutter/command/commands/TranslateCommand.java]: `trnslate` ?

File: src/main/java/emu/grasscutter/command/commands/TeleportCommand.java [hunk 3]
```diff
@@ -54,27 +54,31 @@ private void teleport(Player player, String x, String y, String z) {
...
+            pos.addZ(getByStr(z));
         } else {
             pos.setZ(Float.parseFloat(z));
         }
 
         player.getScene().broadcastPacket(new PacketSceneEntityAppearNotify(player));
         CommandHandler.sendMessage(player, "Teleport " + player.getUid() + " to " + pos);
     }
+
+    private float getByStr(String s)
```
Comment (alphagocc) [inline src/main/java/emu/grasscutter/command/commands/TeleportCommand.java]: get what by str?

File: src/main/java/emu/grasscutter/command/commands/TranslateCommand.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                }
+        }
+    }
+
+    private void translate(Player player, String disntance) {
+        Position pos = player.getPos();
+        double dist = Double.parseDouble(disntance);
+        double angle = Math.toRadians(player.getRotation().getY());
+
+        pos.addX((float) (dist * Math.sin(angle)));
```
Comment (alphagocc) [inline src/main/java/emu/grasscutter/command/commands/TranslateCommand.java:57]: Position has a method `translate`

**Review by alphagocc — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (alphagocc) [reply README.md:137]: https://github.com/Grasscutters/Grasscutter/blob/3b62ee67c0c51585c4cbdd153c063aa1e77436d0/src/main/java/emu/grasscutter/utils/Position.java#L122

seems from here, but also makes me confused

File: src/main/java/emu/grasscutter/command/commands/TranslateCommand.java (hunk 2 above)
Comment (QiE2035) [reply src/main/java/emu/grasscutter/command/commands/TranslateCommand.java]: sorry.....just a wrong input....

File: src/main/java/emu/grasscutter/command/commands/TranslateCommand.java (hunk 4 above)
Comment (QiE2035) [reply src/main/java/emu/grasscutter/command/commands/TranslateCommand.java:57]: The 'translate' in Position use the x and y. But it needs x and z...

File: README.md (hunk 1 above)
Comment (QiE2035) [reply README.md:137]: Maybe my command name is not suitable......

File: src/main/java/emu/grasscutter/command/commands/TranslateCommand.java (hunk 4 above)
Comment (alphagocc) [reply src/main/java/emu/grasscutter/command/commands/TranslateCommand.java:57]: Interesting method, I even didn't notice it

**Pull request closed without merging**